File: 20-valid-parentheses/valid-parentheses.go
func isValid(s string) bool {
    stack := []rune{}
    p := map[rune]rune{
        '(':')',
        '{':'}',
        '[':']',
    }
    for _,val := range s {
       if _, exist := p[val]; exist {
            stack = append(stack,val)
       } else {
            if len(stack)==0 || val != p[stack[len(stack)-1]]{
                return false
            }
            stack = stack[:len(stack)-1]
       }
    }
    return len(stack)==0
}

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        // Solution 1 : Stack
        Stack<Character> stack = new Stack<>();
        stack.push(s.charAt(0));
        for (int i=1;i<s.length();i++){
            char open = ' ';
            if(!stack.empty()){
                open=stack.peek();
            }
            char close=s.charAt(i);
            if ((open=='(')&&(close==')')){
                stack.pop();
            }else if ((open=='[')&&(close==']')){
                stack.pop();
            }else if ((open=='{')&&(close=='}')){
                stack.pop();
            }else{
                stack.push(s.charAt(i));
            }
        }
        return stack.empty();

    }
}